File: src/ServerPulse.Frontend/src/app/modules/shared/domain/dtos/statistics/response/statistics/loadMethodStatisticsResponse.ts
import { BaseStatisticsResponse, convertBaseResponse } from "../../../../../index";

export interface LoadMethodStatisticsResponse extends BaseStatisticsResponse {
    getAmount: number;
    postAmount: number;
    putAmount: number;
    patchAmount: number;
    deleteAmount: number;
}
export function convertToLoadMethodStatisticsResponse(data: any): LoadMethodStatisticsResponse {
    return {
        ...convertBaseResponse(data),
        getAmount: data?.GetAmount ?? 0,
        postAmount: data?.PostAmount ?? 0,
        putAmount: data?.PutAmount ?? 0,
        patchAmount: data?.PatchAmount ?? 0,
        deleteAmount: data?.DeleteAmount ?? 0
    };
}

File: src/ServerPulse.Backend/AnalyzerApi/Domain/Dtos/Responses/LoadMethodStatisticsResponse.cs
namespace AnalyzerApi.Domain.Dtos.Responses
{
    public class LoadMethodStatisticsResponse : BaseStatisticsResponse
    {
        public int GetAmount { get; set; }
        public int PostAmount { get; set; }
        public int PutAmount { get; set; }
        public int PatchAmount { get; set; }
        public int DeleteAmount { get; set; }
    }
}